test/format/config_compat.c

```diff
@@ -0,0 +1,92 @@
+  "checkpoint_wait=", "checkpoint.wait", "checkpoints=", "checkpoint", "checksum=", "disk.checksum",
+  "chunk_size=", "lsm.chunk_size", "compaction=", "ops.compaction", "compression=",
+  "btree.compression", "data_extend=", "disk.data_extend", "data_source=", "runs.source",
+  "delete_pct=", "ops.pct.delete", "dictionary=", "btree.dictionary", "direct_io=",
+  "disk.direct_io", "encryption=", "disk.encryption", "evict_max=", "cache.evict_max", "file_type=",
+  "runs.type", "firstfit=", "disk.firstfit", "huffman_key=", "btree.huffman_key", "huffman_value=",
+  "btree.huffman_value", "in_memory=", "runs.in_memory", "independent_thread_rng=",
+  "format.independent_thread_rng", "insert_pct=", "ops.pct.insert", "internal_key_truncation=",
+  "btree.internal_key_truncation", "internal_page_max=", "btree.internal_page_max", "isolation=",
+  "transaction.isolation", "key_gap=", "btree.key_gap", "key_max=", "btree.key_max", "key_min=",
+  "btree.key_min", "leaf_page_max=", "btree.leaf_page_max", "leak_memory=",
+  "wiredtiger.leak_memory", "logging_archive=", "logging.archive", "logging_compression=",
+  "logging.compression", "logging_file_max=", "logging.file_max", "logging_prealloc=",
+  "logging.prealloc", "lsm_worker_threads=", "lsm.worker_threads", "major_timeout=",
+  "format.major_timeout", "memory_page_max=", "btree.memory_page_max", "merge_max=",
+  "lsm.merge_max", "mmap=", "disk.mmap", "mmap_all=", "runs.mmap_all", "modify_pct=",
+  "ops.pct.modify", "ops=", "runs.ops", "prefix_compression=", "btree.prefix_compression",
+  "prefix_compression_min=", "btree.prefix_compression_min", "prepare=", "ops.prepare",
+  "random_cursor=", "ops.random_cursor", "read_pct=", "ops.pct.read", "rebalance=", "ops.rebalance",
+  "repeat_data_pct=", "btree.repeat_data_pct", "reverse=", "btree.reverse", "rows=", "runs.rows",
+  "salvage=", "ops.salvage", "split_pct=", "btree.split_pct", "statistics=", "statistics",
+  "statistics_server=", "statistics.server", "threads=", "runs.threads", "timer=", "runs.timer",
+  "timing_stress_aggressive_sweep=", "stress.aggressive_sweep", "timing_stress_checkpoint=",
+  "stress.checkpoint", "timing_stress_hs_sweep=", "stress.hs_sweep", "timing_stress_split_1=",
+  "stress.split_1", "timing_stress_split_2=", "stress.split_2", "timing_stress_split_3=",
+  "stress.split_3", "timing_stress_split_4=", "stress.split_4", "timing_stress_split_5=",
+  "stress.split_5", "timing_stress_split_6=", "stress.split_6", "timing_stress_split_7=",
+  "stress.split_7", "timing_stress_split_8=", "stress.split_8", "transaction-frequency=",
+  "transaction.frequency", "transaction_timestamps=", "transaction.timestamps", "truncate=",
+  "ops.truncate", "value_max=", "btree.value_max", "value_min=", "btree.value_min", "verify=",
+  "ops.verify", "wiredtiger_config=", "wiredtiger.config", "write_pct=", "ops.pct.write", NULL,
+  NULL,
+};
+
+/*
+ * config_compat --
+ *     Convert old names to new ones.
+ */
+void
+config_compat(const char **origp)
```

Comment:
It would be useful to be able to output old style configuration files as well? Maybe with a command line option? That will give us a way to run an older version of test/format against a failing config to see when a failure started.

Reply:
> It would be useful to be able to output old style configuration files as well? Maybe with a command line option? That will give us a way to run an older version of test/format against a failing config to see when a failure started.

That would be hard to do. Do we do that often enough to do the work?

Alternatively, maybe we could backport format to previous releases? (I suspect that's likely less work.)

Reply:
It's not really releases, so I don't think a backport will help. For example we are using test/format as a reproducer with git bisect for WT-5826. That's looking at older commits, not released versions.

It seems like you've got a mapping from old->new, could you just generate the inverse? It doesn't need to be maintained indefinitely, so even a shell script that does the inverse mapping would be fine with me.

Reply:
A script would definitely be the only way forward. We don't want to have to maintain anything or add some kind of conversion backward for new config items we add. This compat function is static as of the known configurations at this point in time.

Reply:
Sure, pushed in a3d1cbc81.